# Таблица умножения
# Выполнили: Осадчая Татьяна, Игнатова Юлия

# цикл продолжается до тех пор, пока не будет корректно введено имя файла
# пользователь вводит только имя текстового файла, расширение .txt программа подставляет сама
def tabl():
    while (True):
        try:
            name = input ("Введите имя входного файла: ")
            name += ".txt"
            f = open (name, "r")
            break
        except:
            print ('неверный путь к файлу')

# цикл продолжается до тех пор, пока не будет корректно введено имя файла 
# пользователь вводит только имя выходного файла, расширение .txt программа подставляет сама
    while (True):
        try:
             name = input ("Введите имя выходного файла: ")
             name += ".txt"
             f_new = open (name, "w")
             break
        except:
             print ('неверный путь к файлу')

# создание массива, содержащего числа из входного файла
    arr = f.readlines()
    n = len (arr)

# введение числа с клавиатуры
    while True:
        try:
            qq = input("Введите первый множитель: ")
            qq = float (qq)

# проверка корректности введенных данных
        except ValueError: 
            print ('Ошибка ввода, следует ввести число')
            break
            
# производится вывод таблицы на экран и ее запись в выходной файл
        tab = "+" + 19*"-" + "+" + 19*"-" + "+"
        print (tab)
        f_new.write (tab + "\n")
        print ("|  Второй множитель |     Результат     |")
        print (tab)
        f_new.write ("| Второй множитель |   Результат   |"+ "\n")

# цикл для записи второго множителя и результата умножения в ячейки таблицы; для каждого числа создается отдельная строка 
        for i in range (n):
            try:
                arr[i] = float (arr[i])
                s = "|{0:19.2f}|{1:19.2f}|".format(arr[i], arr[i]*qq)
                
# если во входном файле записано не число, а, например, строка, в ячейках таблицы появляется надпись "ошибка"                                 
            except ValueError:
                    l = len (arr[i])
                    s = "|{0:>19}|".format(arr[i].replace('\n', '')) + "             Ошибка|"

# вывод получившихся результатов в таблице на экран     
            print (s)
            print (tab)

# запись получившихся результатов в таблице в выходной файл
            f_new.write (s + "\n")
            f_new.write (tab + "\n")
 
        f.close()
        f_new.close()
        break
    
# запрос повторного ввода
while True:
    tabl()
    print("Вы хотите еще раз запустить программу? (y/n)")
    la = input()
    if la == 'y':
        pass
    elif la == 'n':
        break

# обработка исключения (не введено ни y, ни n)
    else: 
        print("Введено ошибочное значение")
        break
